pulsar/src/commands/config.rs

```diff
@@ -0,0 +1,168 @@
+pub(crate) async fn config(
+    show: bool,
+    chain: Option<String>,
+    farm_size: Option<String>,
+    reward_address: Option<String>,
+    node_dir: Option<String>,
+    farm_dir: Option<String>,
+) -> eyre::Result<()> {
+    // Define the path to your settings.toml file
+    let config_path = parse_config_path()?;
+
+    // if config file doesn't exist, then throw error
+    if !config_path.exists() {
+        bail!("Config file: \"settings.toml\" not found.\nPlease use `pulsar init` command first.");
+    }
+
+    // Load the current configuration
+    let mut config: Config = parse_config()?;
+
+    if show {
+        // Display the current configuration as JSON
+        // Serialize `config` to a pretty-printed JSON string
+        let serialized = serde_json::to_string_pretty(&config)?;
+        println!(
+            "Current Config set as: \n{}\nin file: {:?}",
+            serialized,
+            config_path.to_str().expect("Expected stringified config path")
+        );
+    } else {
+        // no options provided
+        if chain.is_none()
+            && farm_size.is_none()
+            && reward_address.is_none()
+            && node_dir.is_none()
+            && farm_dir.is_none()
+        {
+            println!("At least one option has to be provided.\nTry `pulsar config -h`");
+            return Ok(());
+        }
+
+        // update (optional) the chain
+        if let Some(c) = chain {
+            let new_chain = ChainConfig::from_str(&c)?;
+            if config.chain == new_chain.clone() {
+                bail!("Chain already set");
+            }
+
+            config.chain = new_chain.clone();
+
+            // if chain is changed, then wipe everything (farmer, node, summary) except
+            // config file
+            if parse_config()?.chain == new_chain {
+                wipe(true, true, true, false).await.expect("Error while wiping.");
+            }
+        }
+
+        // update (optional) the farm size
+        if let Some(f) = farm_size {
+            let farm_size = size_parser(&f)?;
+            config.farmer.farm_size = farm_size;
+        }
+
+        // update (optional) the reward address
+        if let Some(r) = reward_address {
+            let reward_address = reward_address_parser(&r)?;
+            config.farmer.reward_address = reward_address;
+        }
+
+        // update (optional) the node directory
+        if let Some(n) = node_dir {
+            let node_dir = PathBuf::from_str(&n).expect("Invalid node directory");
+            create_or_move_data(config.node.directory.clone(), node_dir.clone())?;
+            config.node.directory = node_dir;
+        }
+
+        // update (optional) the farm directory
+        if let Some(fd) = farm_dir {
+            let farm_dir = PathBuf::from_str(&fd).expect("Invalid farm directory");
+            create_or_move_data(config.farmer.farm_directory.clone(), farm_dir.clone())?;
+            if farm_dir.exists() {
```

Comment:
Why would it not exists? Since you are creating it just above.

Reply:
I am able to realize after writing tests for `create_or_move_data` that it's redundant check.
Fixed in this [commit](https://github.com/subspace/pulsar/commit/96e408b430aa1b3316aa06762e2d88ced2f776fa).